### PR TITLE
org.KekikAkademi.BTKSorgu

### PR DESCRIPTION
**Pull request by keyiflerolsun**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

File: org.KekikAkademi.BTKSorgu.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+modules:
+- python3-tkinter.yaml
+- tesseract.yaml
+- python3-requirements.json
+
+- name: BTKSorgu-yukleyici
+  buildsystem: simple
+  build-options:
+    build-args:
+      - "--share=network"
```
Comment (hfiguiere) [inline org.KekikAkademi.BTKSorgu.yml:33]: this will not work, it's not allowed. The build must be performed offline.

File: tesseract.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    dest: traineddata
+  # Tweak location of tessdata folder
+  # https://src.fedoraproject.org/rpms/tesseract/blob/7ae0e18c8f30353dea96807a4a445d1b3aba8d74/f/tesseract_datadir.patch
+  - type: patch
+    path: tesseract-4.1.1-datadir.patch
+post-install:
+  - install -p -D -m 0644 "traineddata"/*.traineddata -t "${FLATPAK_DEST}/share/tesseract/tessdata/";
+  - install -p -D -m 0644 "traineddata/script"/*.traineddata -t "${FLATPAK_DEST}/share/tesseract/tessdata/script/";
+modules:
+  - leptonica.yaml
```
Comment (hfiguiere) [inline tesseract.yaml:39]: this is missing. Note that having separate manifest is not a good practice. This applies to the place you copied this from.

File: tesseract.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  #  dest: traineddata
+  - type: git
+    url: https://github.com/tesseract-ocr/tessdata_best.git
+    #tag: 4.0.0
+    commit: 9e8aeef07ce8c4f6e6519577cee76363246bc6de
+    dest: traineddata
+  # Tweak location of tessdata folder
+  # https://src.fedoraproject.org/rpms/tesseract/blob/7ae0e18c8f30353dea96807a4a445d1b3aba8d74/f/tesseract_datadir.patch
+  - type: patch
+    path: tesseract-4.1.1-datadir.patch
```
Comment (hfiguiere) [inline tesseract.yaml:34]: this patch is missing.

File: leptonica.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+name: leptonica
```
Comment (hfiguiere) [inline leptonica.yaml:1]: this should not be a separate manifest.

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

File: SRC/leptonica.yaml
Comment (hfiguiere) [inline SRC/leptonica.yaml:1]: I will ask again: this should NOT be a separate manifest.

File: SRC/tesseract.yaml
Comment (hfiguiere) [inline SRC/tesseract.yaml:1]: this should NOT be a separate manifest.

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

File: org.KekikAkademi.BTKSorgu.yml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+runtime-version: "22.08"
+sdk: org.freedesktop.Sdk
+command: BTKSorguGUI
+
+finish-args:
+  # ? X11 + XShm Erişimi
+  - --share=ipc
+  # - --socket=x11
+  # ? Wayland Erişimi
+  - --socket=wayland
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.KekikAkademi.BTKSorgu.yml:14]: you didn't answer the question.

[the remark this answers is not in the document]
Comment (keyiflerolsun) [reply org.KekikAkademi.BTKSorgu.yml:14]: Tcl/Tk

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

File: org.KekikAkademi.BTKSorgu.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (keyiflerolsun) [reply org.KekikAkademi.BTKSorgu.yml:14]: i don't have wayland system

File: SRC/tesseract.yaml
Comment (hfiguiere) [inline SRC/tesseract.yaml:1]: as I mentionned previously this should be part of the main manifest too since it is not autogenerated.

Comment (keyiflerolsun) [reply SRC/tesseract.yaml:1]: fix test : https://github.com/flathub/flathub/pull/4158/commits/a8e51fc14da35f065cd057db21400d45a3cf926c

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

File: SRC/tesseract.yaml
Comment (keyiflerolsun) [reply SRC/tesseract.yaml:1]: probably second fix is done : https://github.com/flathub/flathub/pull/4158/commits/8dc832df7ec51620637e502ef181d28dd6120473

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
> Build [41692 successful](https://buildbot.flathub.org/#/builders/6/builds/41692) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/24329/org.KekikAkademi.BTKSorgu.flatpakref
> ```

This is work <3 
image

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
> Build [41712 failed](https://buildbot.flathub.org/#/builders/6/builds/41712)

`fatal: unable to access 'https://github.com/flathub/buildbot-config.git/': gnutls_handshake() failed: The TLS connection was non-properly terminated.`

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
> Build [41726 successful](https://buildbot.flathub.org/#/builders/6/builds/41726) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/24355/org.KekikAkademi.BTKSorgu.flatpakref
> ```

i think ready to publish 
image

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

File: org.KekikAkademi.BTKSorgu.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.KekikAkademi.BTKSorgu.yml:14]: I can tell you this won't work as tk doesn't support wayland.

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

File: org.KekikAkademi.BTKSorgu.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (keyiflerolsun) [reply org.KekikAkademi.BTKSorgu.yml]: i remove it

**Comment by keyiflerolsun** [comment]
bot, build org.KekikAkademi.BTKSorgu

**Pull request closed without merging**